plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -146,14 +146,16 @@ func (a *Elasticsearch) Write(metrics []telegraf.Metric) error {
 		m := make(map[string]interface{})
 		mfields := make(map[string]interface{})
 
+		// Elasticsearch/java/json quirks for correct field mapping
+		// value as integer cannot be bigger than MaxInt64 or smaller than MinInt64; as a float it cannot be bigger than float32
 		for k, v := range metric.Fields() {
-			switch x := v.(type) {
-			// Truncate values too big/small for Elasticsearch to not complain when creating a dynamic field mapping
-			case float64:
-				if (x > 0 && x > math.MaxFloat32) || (x < 0 && x < math.SmallestNonzeroFloat32) {
-					v = float32(x)
-					log.Printf("W! Elasticsearch output metric %s truncated (value %v is too big or too small, truncating to %v)", k, x, v)
-				}
+			switch {
+			case v.(float64) > math.MaxInt64:
```

Comment:
does this work if `v` is a `float32`?

Not sure I understand how this compiles, since it's comparing a float64 to an int64?